src/renderers/dom/stack/client/ReactDOMComponent.js

```diff
@@ -583,6 +584,20 @@ ReactDOMComponent.Mixin = {
         );
         didWarnShadyDOM = true;
       }
+
+      if (__DEV__) {
+        if (this._namespaceURI === DOMNamespaces.html) {
```

Comment:
I see @gaearon did [mention previously](https://github.com/facebook/react/issues/3726#issuecomment-262009657) he'd be OK with that